src/cluster_asm.c

```diff
@@ -1325,11 +1332,14 @@ void clusterSyncSlotsCommand(client *c) {
     /* Only internal clients are allowed to execute this command to avoid
      * potential attack, since some state changes are not well protected,
      * external clients may damage the slot migration state. */
-    if (!(c->flags & CLIENT_INTERNAL)) {
+    if (!(c->flags & (CLIENT_INTERNAL | CLIENT_MASTER))) {
         addReplyError(c, "CLUSTER SYNCSLOTS subcommands are only allowed for internal clients");
         return;
     }
 
+    /* Only allow CONF subcommand on replica. */
+    if (server.masterhost && strcasecmp(c->argv[2]->ptr, "conf")) return;
```

Comment:
should we reply something so it would be easier to detect the bug on the sender side?

Reply:
since now the source proxy the importing command stream, i am worried the source may also replicate some `cluster syncslots xxx` command.

But so far, i didn't find there are other `cluster syncslots xxx` command except `cluster syncslots conf`, the reason i guess is that the source sends `cluster syncslots conf` and `restore` command together, but for others, like `cluster syncslots  snapshot-eof` and `cluster syncslots  stream-eof`, actually this is the last command of that client, we remove `CLIENT_MASTER` flag and close the client after executing command, so we don't have chance to replicate them.

i am not sure if we could introduce other subcommands, so i keep this check, and this check is defensive, and ignore if it happens

Reply:
maybe `cluster syncslots stream-eof` can arrive on the replicas, since we do not close clients if the destination received `stream-eof` command during streaming buffer into DB. that is correct path. so we should not reply

Reply:
okay, make sense